legend-engine-xt-mastery-grammar/src/main/java/org/finos/legend/engine/language/pure/dsl/mastery/compiler/toPureGraph/HelperMasterRecordDefinitionBuilder.java

```diff
@@ -120,9 +116,207 @@ public static RichIterable<Root_meta_pure_mastery_metamodel_RecordSource> buildR
+                {
+                    parentClass = (Class<?>) parentType;
+                }
+                else
+                {
+                    parentClass = null;
+                }
+            }
+            return purePrecedenceRule;
+        }
+
+        private Root_meta_pure_mastery_metamodel_precedence_SourcePrecedenceRule visitSourcePrecedenceRule(PrecedenceRule precedenceRule)
+        {
+            SourcePrecedenceRule sourcePrecedenceRule = (SourcePrecedenceRule) precedenceRule;
+            Root_meta_pure_mastery_metamodel_precedence_SourcePrecedenceRule_Impl pureSourcePrecedenceRule = new Root_meta_pure_mastery_metamodel_precedence_SourcePrecedenceRule_Impl("");
+            pureSourcePrecedenceRule._precedence(sourcePrecedenceRule.precedence);
+            String ACTION_TYPE_FULL_PATH = MASTERY_PACKAGE_PREFIX + "::precedence::RuleAction";
+            pureSourcePrecedenceRule._action(context.resolveEnumValue(ACTION_TYPE_FULL_PATH, sourcePrecedenceRule.action.name()));
+            return pureSourcePrecedenceRule;
+        }
+
+        private Root_meta_pure_mastery_metamodel_precedence_ConditionalRule visitConditionalRule(PrecedenceRule precedenceRule)
+        {
+            Root_meta_pure_mastery_metamodel_precedence_ConditionalRule pureConditionalRule = new Root_meta_pure_mastery_metamodel_precedence_ConditionalRule_Impl("");
+            ConditionalRule conditionalRule = (ConditionalRule) precedenceRule;
+            validatePredicateInput(conditionalRule.predicate);
+            pureConditionalRule._predicate(HelperValueSpecificationBuilder.buildLambda(conditionalRule.predicate, context));
+            return pureConditionalRule;
+        }
+
+        private void validatePredicateInput(Lambda predicate)
+        {
+            validateInputVariableNames(predicate);
+            validateInputVariableType(predicate);
+            validateInputMultiplicity(predicate);
+        }
+
+        private void validateInputVariableNames(Lambda predicate)
+        {
+            Set<String> actualNames = predicate.parameters.stream().map(variable -> variable.name).collect(Collectors.toSet());
+            if (!CONDITIONAL_BLOCK_RULE_PREDICATE_INPUT.equals(actualNames))
+            {
+                throw new EngineException(
+                        format("Incorrect input variable names, expect %s but received %s", CONDITIONAL_BLOCK_RULE_PREDICATE_INPUT, actualNames),
+                        EngineErrorType.COMPILATION);
+            }
+        }
+
+        private void validateInputVariableType(Lambda predicate)
+        {
+            Set<String> inputVariableTypes = predicate.parameters.stream().map(variable -> variable._class).collect(Collectors.toSet());
+            if (inputVariableTypes.size() != 1)
+            {
+                throw new EngineException(format("Two of the same master Record Classes are expected, received: %s", inputVariableTypes), EngineErrorType.COMPILATION);
```

Comment:
On the errors messages for these exceptions, can we be consistent on how we output "Master Record". Some places have lower case, upper case or a mix.

Reply:
Thanks, this has been resolved